include/lsst/afw/table/io/Persistable.h

```diff
@@ -203,8 +203,19 @@ class PersistableFacade {
      *                           afw::fits::DEFAULT_HDU skips the primary HDU if it is empty.
      */
     static std::shared_ptr<T> readFits(fits::MemFileManager& manager, int hdu = fits::DEFAULT_HDU) {
-        return std::dynamic_pointer_cast<T>(Persistable::_readFits(manager, hdu));
+        return dynamicCast(Persistable::_readFits(manager, hdu));
     }
+
+    /**
+     * Dynamically cast a shared_ptr
+     *
+     * You must provide an explicit template instantiation in the .cc file
+     * for each class that inherits from PersistableFacade.
+     * Designed to work around RTTI issues on macOS with hidden symbols;
+     *
+     * @throw lsst::pex::exceptions::LogicError if the cast fails
+     */
+    static std::shared_ptr<T> dynamicCast(std::shared_ptr<Persistable> const &ptr);
```

Comment:
Why is this method part of `PersistableFacade`? Can you provide a link for the "issues"?

Reply:
Why throw `LogicError` instead of `std::bad_cast` or, if you want an LSST exception, `pex::exceptions::TypeError`?

Reply:
This can only occur if the code has an internal bug so I thought `LogicError` seemed most appropriate to me.

We never even bothered to test it before, since it always worked. But symbol visibility issues broke it and the error was hard to diagnose down due to the lack of a test.